# Do not leak a string pointer
#
# Copyright (C) 2006 - 2009 Nokia Corporation.
# This file is distributed under the terms of GNU LGPL license, either version 2
# of the License, or (at your option) any later version.
#
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine.c gtkhtml-3.24.4/gtkhtml/htmlengine.c
--- gtkhtml-3.24.4/gtkhtml/htmlengine.c	2007-02-15 12:41:23.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine.c	2007-02-15 12:46:07.000000000 +0200
@@ -5251,7 +5251,8 @@
 	e->formTextArea = NULL;
 	e->inOption = FALSE;
 	e->inTextArea = FALSE;
-	e->formText = g_string_new ("");
+	if (NULL == e->formText)
+		e->formText = g_string_new ("");
 
 	e->flow = NULL;
 
